pool/pool.go

```diff
@@ -505,17 +554,35 @@ func (c *clientStatusMonitor) address() string {
 	return c.addr
 }
 
-func (c *clientStatusMonitor) errorRate() uint32 {
-	return c.errorCount.Load()
+func (c *clientStatusMonitor) incErrorRate() {
+	c.currentErrorCount.Inc()
+	c.overallErrorCount.Inc()
+}
+
+func (c *clientStatusMonitor) currentErrorRate() uint32 {
+	return c.currentErrorCount.Load()
+}
+
+func (c *clientStatusMonitor) overallErrorRate() uint64 {
+	return c.overallErrorCount.Load()
 }
 
 func (c *clientStatusMonitor) resetErrorCounter() {
-	c.errorCount.Store(0)
+	c.currentErrorCount.Store(0)
+}
+
+func (c *clientStatusMonitor) latency() time.Duration {
+	return time.Duration(c.allTime.Load() / c.allRequests.Load())
+}
+
+func (c *clientStatusMonitor) incRequests(elapsed time.Duration) {
+	c.allTime.Add(uint64(elapsed))
+	c.allRequests.Inc()
```

Comment:
While 2 atomics make sense, now we have 5, may be we it is better to use a single mutex for all of them?